pkg/controller/che/che_controller.go

```diff
@@ -761,128 +764,17 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
-				}
-			} else {
-				// create Keycloak route
-				route, err := deploy.SyncRouteToCluster(deployContext, "keycloak", "", "keycloak", 8080)
-				if !tests {
-					if route == nil {
-						logrus.Info("Waiting on route 'keycloak' to be ready")
-						if err != nil {
-							logrus.Error(err)
-						}
-
-						return reconcile.Result{RequeueAfter: time.Second * 1}, err
-					}
-
-					keycloakURL := protocol + "://" + route.Spec.Host
-					if instance.Spec.Auth.IdentityProviderURL != keycloakURL {
-						instance.Spec.Auth.IdentityProviderURL = keycloakURL
-						if err := r.UpdateCheCRSpec(instance, "Keycloak URL", keycloakURL); err != nil {
-							instance, _ = r.GetCR(request)
-							return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 1}, err
-						}
-						instance.Status.KeycloakURL = keycloakURL
-						if err := r.UpdateCheCRStatus(instance, "status: Keycloak URL", keycloakURL); err != nil {
-							instance, _ = r.GetCR(request)
-							return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 1}, err
-						}
-					}
-				}
-			}
-
-			deploymentStatus := deploy.SyncKeycloakDeploymentToCluster(deployContext)
-			if !tests {
-				if !deploymentStatus.Continue {
-					logrus.Info("Waiting on deployment 'keycloak' to be ready")
-					if deploymentStatus.Err != nil {
-						logrus.Error(deploymentStatus.Err)
-					}
-
-					return reconcile.Result{Requeue: deploymentStatus.Requeue}, deploymentStatus.Err
-				}
-			}
-
-			if !tests {
-				if !instance.Status.KeycloakProvisoned {
-					if err := deploy.ProvisionKeycloakResources(deployContext); err != nil {
-						logrus.Error(err)
-						return reconcile.Result{RequeueAfter: time.Second}, err
-					}
-
-					for {
-						instance.Status.KeycloakProvisoned = true
-						if err := r.UpdateCheCRStatus(instance, "status: provisioned with Keycloak", "true"); err != nil &&
-							errors.IsConflict(err) {
-							instance, _ = r.GetCR(request)
-							continue
-						}
-						break
-					}
-				}
-			}
-
-			if isOpenShift {
-				doInstallOpenShiftoAuthProvider := instance.Spec.Auth.OpenShiftoAuth
-				if doInstallOpenShiftoAuthProvider {
-					openShiftIdentityProviderStatus := instance.Status.OpenShiftoAuthProvisioned
-					if !openShiftIdentityProviderStatus {
-						if err := r.CreateIdentityProviderItems(instance, request, cheFlavor, deploy.KeycloakDeploymentName, isOpenShift4); err != nil {
-							return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
-						}
-					}
-				}
+	provisioned, err := deploy.SyncIdentityProviderToCluster(deployContext, cheHost, protocol, cheFlavor)
+	if !tests {
+		if !provisioned {
+			if err != nil {
+				logrus.Errorf("Error provisioning the identity provider to cluster: %v", err)
 			}
+			return reconcile.Result{RequeueAfter: time.Second * 1}, err
 		}
 	}
 
-	provisioned, err := deploy.SyncDevfileRegistryToCluster(deployContext)
+	provisioned, err = deploy.SyncDevfileRegistryToCluster(deployContext, cheHost)
```

Comment:
`cheHost` variable is defined below. So, you always pass `""` here.
Am I right?

Reply:
I think `cheHost` is declared on https://github.com/eclipse/che-operator/pull/378/files/62facaa4dfe0b3f31a8de36f2aa79c65c32d7469#diff-d9445e614f0d728fd140771b0fd29c50R721 and first assigned on https://github.com/eclipse/che-operator/pull/378/files/62facaa4dfe0b3f31a8de36f2aa79c65c32d7469#diff-d9445e614f0d728fd140771b0fd29c50R733 for Kubernetes and on https://github.com/eclipse/che-operator/pull/378/files/62facaa4dfe0b3f31a8de36f2aa79c65c32d7469#diff-d9445e614f0d728fd140771b0fd29c50R752 for OpenShift.

I think you're referring to https://github.com/eclipse/che-operator/pull/378/files/62facaa4dfe0b3f31a8de36f2aa79c65c32d7469#diff-d9445e614f0d728fd140771b0fd29c50R864. I think that assignment is superfluous because the `instance.Spec.Server.CheHost` has been set on the lines mentioned above to be the same as `cheHost`. The control flow is a little bit confusing in that super massive method though so I may be wrong...

Reply:
Agree. Thank you for explanation.